### ISSUE TITLE
Generating and seding UserEvent

### ISSUE THREAD

**Issue by alejandro-aznar** [asterisk-java/asterisk-java#268]
Hi,

I have been looking around for quite a while and I'm unable to find a way to generate and send a UserEvent.

The constructor of the class requires another UserEvent to be created, which confuses me quite a lot. Maybe someone has some working example?

Or maybe there is a way to directly send whatever you want to the AMI?

Thanks in advance

Extra info:

I want to create a UserEvent then send it using userEventAction. And maybe get the response, although not 100% necessary.

Later do it with a custom one, registering the class to the managerconnection and so on.

Example with asterisk code would be: UserEvent(VIPCall|firstName: Jon);

It seems to me like the documentation is confusing or lacking on that interaction or atleast personally I haven't been able to figure this out =/

**Comment by alejandro-aznar** [comment]
We managed to create a UserEvent by simply using a String as the parameter, which creates an Attribute "source: paramString"

However, we found a bigger issue. When sending the UserEvent using UserEventAction and the manager.sendAction, the Attribute "UserEvent: nameofevent" is always lowecase. (also the added parameter names in case of a custom UserEvent)

I'm not really sure of why it is simplemented in that way, forcing the user event attribute value to be always lower case but that makes it impossible to catch the event with our other service that uses a eventuser value with all caps.

Is there any workaround without having to fork java-asterisk or our other service ? For now we will just use a socket to send that part.

Thanks.

**Comment by seanbright** [comment]
Yes. This is poorly implemented. I will see if we can make it better.
